components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -706,54 +878,92 @@ void SolanaTxManager::OnGetAccountInfo(
   }
-
-  auto chain_id = meta->chain_id();
-  GetSolanaBlockTracker()->GetLatestBlockhash(
-      chain_id,
-      base::BindOnce(&SolanaTxManager::OnGetLatestBlockhashForGetEstimatedTxFee,
-                     weak_ptr_factory_.GetWeakPtr(), std::move(meta),
-                     std::move(callback)),
-      true);
 }
 
 void SolanaTxManager::OnGetLatestBlockhashForGetEstimatedTxFee(
     std::unique_ptr<SolanaTxMeta> meta,
-    GetEstimatedTxFeeCallback callback,
+    GetEstimatedTxBaseFeeCallback callback,
     const std::string& latest_blockhash,
     uint64_t last_valid_block_height,
     mojom::SolanaProviderError error,
     const std::string& error_message) {
   if (error != mojom::SolanaProviderError::kSuccess) {
-    std::move(callback).Run(0, error, error_message);
+    std::move(callback).Run({}, 0, error, error_message);
     return;
   }
 
+  const std::string& chain_id = meta->chain_id();
   meta->tx()->message()->set_recent_blockhash(latest_blockhash);
   meta->tx()->message()->set_last_valid_block_height(last_valid_block_height);
   const std::string base64_encoded_message =
       meta->tx()->GetBase64EncodedMessage();
   json_rpc_service_->GetSolanaFeeForMessage(
-      meta->chain_id(), base64_encoded_message,
+      chain_id, base64_encoded_message,
       base::BindOnce(&SolanaTxManager::OnGetFeeForMessage,
-                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback),
+                     std::move(meta)));
 }
 
-void SolanaTxManager::OnGetFeeForMessage(GetEstimatedTxFeeCallback callback,
+void SolanaTxManager::OnGetFeeForMessage(GetEstimatedTxBaseFeeCallback callback,
+                                         std::unique_ptr<SolanaTxMeta> meta,
                                          uint64_t tx_fee,
                                          mojom::SolanaProviderError error,
                                          const std::string& error_message) {
-  std::move(callback).Run(tx_fee, error, error_message);
+  std::move(callback).Run(std::move(meta), tx_fee, error, error_message);
+}
+
+void SolanaTxManager::GetSolanaTxFeeEstimation(
+    const std::string& chain_id,
+    const std::string& tx_meta_id,
+    GetSolanaTxFeeEstimationCallback callback) {
+  // Get the TxMeta.
+  std::unique_ptr<SolanaTxMeta> meta =
+      GetSolanaTxStateManager()->GetSolanaTx(tx_meta_id);
+  if (!meta) {
+    std::move(callback).Run(
+        {}, mojom::SolanaProviderError::kInternalError,
+        l10n_util::GetStringUTF8(IDS_BRAVE_WALLET_TRANSACTION_NOT_FOUND));
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&SolanaTxManager::FinishGetSolanaTxFeeEstimation,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  GetSolanaTxFeeEstimationAndMeta(chain_id, std::move(meta),
+                                  std::move(internal_callback));
+}
+
+void SolanaTxManager::GetSolanaTxFeeEstimationAndMeta(
```

Comment:
We should have a comment in this function to describe the whole flow (process) and what's our expected behavior for all cases, otherwise it's hard to see the whole picture with things scatter around.
Also, GetSolanaTxFeeEstimationAndMeta seems a bit misleading as we're not really getting meta but just something that's passed in.